modules/client/src/main/java/org/apache/ignite/internal/jdbc/ConnectionPropertiesImpl.java

```diff
@@ -0,0 +1,757 @@
+        "Schema name of the connection", "PUBLIC", null, false, null);
+
+    /** Query timeout. */
+    private final IntegerProperty qryTimeout = new IntegerProperty("queryTimeout",
+        "Sets the number of seconds the driver will wait for a <code>Statement</code> object to execute." +
+            " Zero means there is no limits.",
+        null, false, 0, Integer.MAX_VALUE);
+
+    /** JDBC connection timeout. */
+    private final IntegerProperty connTimeout = new IntegerProperty("connectionTimeout",
+        "Sets the number of milliseconds JDBC client will waits for server to response." +
+            " Zero means there is no limits.",
+        0L, false, 0, Integer.MAX_VALUE);
+
+    /** Properties array. */
+    private final ConnectionProperty[] propsArray = {qryTimeout, connTimeout};
+
+    /** {@inheritDoc} */
+    @Override public String getSchema() {
+        return schema.value();
+    }
+
+    /** {@inheritDoc} */
+    @Override public void setSchema(String schema) {
+        this.schema.setValue(schema);
+    }
+
+    /** {@inheritDoc} */
+    @Override public String getUrl() {
+        if (url != null)
+            return url;
+        else {
+            if (ArrayUtils.nullOrEmpty(getAddresses()))
+                return null;
+
+            StringBuilder sbUrl = new StringBuilder(URL_PREFIX);
+
+            HostAndPortRange[] addrs = getAddresses();
+
+            for (int i = 0; i < addrs.length; i++) {
+                if (i > 0)
+                    sbUrl.append(',');
+
+                sbUrl.append(addrs[i].toString());
+            }
+
+            if (!isEmpty(getSchema()))
+                sbUrl.append('/').append(getSchema());
+
+            return sbUrl.toString();
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public void setUrl(String url) throws SQLException {
+        this.url = url;
+
+        init(url, new Properties());
+    }
+
+    /** {@inheritDoc} */
+    @Override public HostAndPortRange[] getAddresses() {
+        return addrs;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void setAddresses(HostAndPortRange[] addrs) {
+        this.addrs = addrs;
+    }
+
+    /** {@inheritDoc} */
+    @Override public Integer getQueryTimeout() {
+        return qryTimeout.value();
+    }
+
+    /** {@inheritDoc} */
+    @Override public void setQueryTimeout(@Nullable Integer timeout) throws SQLException {
+        qryTimeout.setValue(timeout);
+    }
+
+    /** {@inheritDoc} */
+    @Override public int getConnectionTimeout() {
+        return connTimeout.value();
+    }
+
+    /** {@inheritDoc} */
+    @Override public void setConnectionTimeout(@Nullable Integer timeout) throws SQLException {
+        connTimeout.setValue(timeout);
+    }
+
+    /**
+     * Init connection properties.
+     *
+     * @param url URL connection.
+     * @param props Environment properties.
+     * @throws SQLException On error.
+     */
+    public void init(String url, Properties props) throws SQLException {
```

Comment:
```suggestion
    private void init(String url, Properties props) throws SQLException {
        assert props != null;
```

Reply:
I am not sure about this suggestion. init() method could produce SQLException and cannot be put in the constructor. I suggest leaving it public.